File: src/parser/parsers/newline.js
"use strict";

const node = require("../helpers/nodes.js");
const add = require("../helpers/tree-add.js");

/**
 * ----------------------------------------------------------- Parsing Breakdown
 * \n
 * ^-Newline character.
 * -----------------------------------------------------------------------------
 *
 * @param  {object} S - State object.
 * @return {undefined} - Nothing is returned.
 */
module.exports = (S) => {
	let N = node(S, "NEWLINE");

	N.start = N.end = S.i;

	S.line++;
	S.column = 0;
	S.sol_char = "";

	add(S, N);
};

File: src/parser/nim/parser/parsers/newline.nim
import ../helpers/[tree_add, types, charsets]

# ------------------------------------------------------------ Parsing Breakdown
# \n
# ^-Newline character.
# ------------------------------------------------------------------------------

# @param  {object} S - State object.
# @return Nothing is returned.
proc p_newline*(S: State) =
    var N = node(nkNewline, S)

    N.start = S.i
    N.stop = S.i

    S.line = S.line + 1
    S.column = 0
    S.sol_char = C_NULLB

    add(S, N)
